### DISCUSSION TITLE
Word-level transcripts from .NET interface?

### DISCUSSION THREAD

**Discussion by williamj77** [deepgram/community#91] [General help] [answered]
Hello,

How do I extract the word-level JSON results using the ,NET interface?

Here is my code:

                using (FileStream fs = File.OpenRead(strAudioFile))
                {
                    var response = await deepgramClient.Transcription.Prerecorded.GetTranscriptionAsync(
                        new Deepgram.Transcription.StreamSource(
                            fs,
                            "audio/wav"),
                        new Deepgram.Transcription.PrerecordedTranscriptionOptions()
                        {
                            Punctuate = true,
                            Utterances = true
                        });

                    //strReults = response.Results //?
                }`

Thanks.

Regards,
William Johnston

**Comment by BrianHillis-DG** [comment, marked as answer]
Hi @williamj77 ,

To get the words array, you can do:
``` Csharp
var words = response.Results.Channels.First().Alternatives.First().Words;
```
which enables something like:
``` Csharp 
foreach(var word in words){
  Console.WriteLine(word.PunctuatedWord);
}
```

Hope this helps! Happy Hacking! 

Best,
Brian
